project/data_analysis/python/get_best_fit_mflike.py

```diff
@@ -39,67 +38,98 @@
 fg_norm = d["fg_norm"]
 fg_params = d["fg_params"]
 fg_components = d["fg_components"]
-fg_dict = best_fits.get_foreground_dict(l_th, freq_list, fg_components, fg_params, fg_norm)
+
+passbands = {}
+do_bandpass_integration = d["do_bandpass_integration"]
+
+narrays, sv_list, ar_list = pspipe_list.get_arrays_list(d)
+for sv, ar in zip(sv_list, ar_list):
+
+    freq_info = d[f"freq_info_{sv}_{ar}"]
+    if do_bandpass_integration:
+        nu_ghz, pb = np.loadtxt(freq_info["passband"]).T
+    else:
+        nu_ghz, pb = np.array([freq_info["freq_tag"]]), np.array([1.])
+
+    passbands[f"{sv}_{ar}"] = [nu_ghz, pb]
+
+fg_dict = best_fits.get_foreground_dict(l_th, passbands, fg_components,
+                                        fg_params, fg_norm,)
+
+spectra_list = pspipe_list.get_spec_name_list(d, char = "_")
 fg= {}
-for freq1 in freq_list:
-    for freq2 in freq_list:
-        fg[freq1, freq2] = {}
+
+for sv1, ar1 in zip(sv_list, ar_list):
+    for sv2, ar2 in zip(sv_list, ar_list):
+        name1 = f"{sv1}_{ar1}"
+        name2 = f"{sv2}_{ar2}"
+        fg[name1, name2] = {}
         for spec in spectra:
-            fg[freq1,freq2][spec] = fg_dict[spec.lower(), "all", freq1, freq2]
-        so_spectra.write_ps(f"{bestfit_dir}/fg_{freq1}x{freq2}.dat", l_th, fg[freq1,freq2], type, spectra=spectra)
+            fg[name1, name2][spec] = fg_dict[spec.lower(), "all", name1, name2]
+
+        so_spectra.write_ps(f"{bestfit_dir}/fg_{name1}x{name2}.dat", l_th, fg[name1, name2], type, spectra=spectra)
 
 for spec in spectra:
     plt.figure(figsize=(12, 12))
-    for freq1 in freq_list:
-        for freq2 in freq_list:
-            name = f"{freq1}x{freq2}_{spec}"
-            cl_th_and_fg = ps_dict[spec]
+    for ps_name in spectra_list:
+        name1, name2 = ps_name.split("x")
+        name = f"{ps_name}_{spec}"
+        cl_th_and_fg = ps_dict[spec]
 
-            if spec == "TT":
-                plt.semilogy()
+        if spec == "TT":
+            plt.semilogy()
+        if spec.lower() in d["fg_components"].keys():
```

Comment:
As far as I understand, the `if/else` block is just to treat TE vs. ET, TB vs. BT and so on. But I wonder if someone pass a list of `fg_components` without TE foregrounds what's going on (there will be a ET foreground but no TE one). I think it will be better to be more explicit here and set 
```python
if spec.lower() in ["tt", "te", "ee", "tb", "bb]:
```